lib/Dialect/SV/Transforms/HWExportModuleHierarchy.cpp

```diff
@@ -68,38 +75,55 @@ static std::string extractHierarchyFromTop(hw::HWModuleOp op,
         printHierarchy(op, symbolTable, J);
     });
   });
-
-  return resultBuffer;
 }
 
 /// Find the modules corresponding to the firrtl mainModule and DesignUnderTest,
 /// and if they exist, emit a verbatim op with the module hierarchy for each.
 void HWExportModuleHierarchyPass::runOnOperation() {
   mlir::ModuleOp mlirModule = getOperation();
-  auto builder = OpBuilder::atBlockEnd(mlirModule.getBody());
   Optional<SymbolTable> symbolTable = None;
+  bool directoryCreated = false;
 
-  bool anythingChanged = false;
   for (auto op : mlirModule.getOps<hw::HWModuleOp>()) {
-    if (auto attr = op->getAttr("firrtl.moduleHierarchyFile")) {
+    if (auto attr = op->getAttrOfType<hw::OutputFileAttr>(
+            "firrtl.moduleHierarchyFile")) {
       if (!symbolTable)
         symbolTable = SymbolTable(mlirModule);
-      auto verbatimOp = builder.create<sv::VerbatimOp>(
-          builder.getUnknownLoc(),
-          extractHierarchyFromTop(op, symbolTable.getValue()));
-      verbatimOp->setAttr("output_file", attr);
-      op->removeAttr("firrtl.moduleHierarchyFile");
-      anythingChanged = true;
+
+      if (!directoryCreated) {
+        auto error = llvm::sys::fs::create_directory(directoryName);
+        if (error) {
+          llvm::errs() << error.message() << "\n";
```

Comment:
instead of printing this message (and the one below) to stderr, please emit it to the op location, e.g. `op->emitError("error generating file in HWExportModuleHierarchy: ") << error.message();`

Reply:
Thanks! Updated in https://github.com/llvm/circt/pull/1931/commits/3ad60b1ca024f7cddbeee355315c01e6446200e9.